### PR TITLE
Check table and view existence in Memory connector

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Check table and view existence in Memory connector

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

**Comment by findepi** [comment]
> (x) No release notes entries required.

True if we merge this in 374, otherwise this is a bug fix.

https://github.com/trinodb/trino/pull/11421#issuecomment-1065164295

**Review by findepi — APPROVED** [review]

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java [hunk 1]
```diff
@@ -115,9 +116,29 @@ public synchronized void dropSchema(ConnectorSession session, String schemaName)
...
+        if (!isSchemaEmpty(schemaName)) {
+            throw new TrinoException(SCHEMA_NOT_EMPTY, "Schema not empty: " + schemaName);
+        }
+
         verify(schemas.remove(schemaName));
     }
 
+    private boolean isSchemaEmpty(String schemaName)
+    {
+        if (tables.values().stream()
```
Comment (hashhar) [inline plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java:129]: Is this thread-safe? It's backed by a normal HashMap and there is no synchronisation.

**Review by hashhar — APPROVED** [review]
LGTM % thread-safety (it was pre-existing anyway so I'm ok with current impl too).

File: plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java:129]: It should be. https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html

> First, it is not possible for two invocations of synchronized methods on the same object to interleave. When one thread is executing a synchronized method for an object, all other threads that invoke synchronized methods for the same object block (suspend execution) until the first thread is done with the object.

**Pull request merged**